instana/w3c_trace_context/treceparent.py

```diff
@@ -0,0 +1,64 @@
+# (c) Copyright IBM Corp. 2021
+# (c) Copyright Instana Inc. 2021
+
+from ..log import logger
+import re
+
+
+class Traceparent:
+    SPECIFICATION_VERSION = "00"
+    TRACEPARENT_REGEX = re.compile("[0-9a-f]{2}-(?!0{32})([0-9a-f]{32})-(?!0{16})([0-9a-f]{16})-[0-9a-f]{2}")
+
+    def validate(self, traceparent):
+        """
+        Method used to validate the traceparent header
+        :param traceparent: string
+        :return: traceparent or None
+        """
+        try:
+            if self.TRACEPARENT_REGEX.match(traceparent) and traceparent.split("-")[0] == self.SPECIFICATION_VERSION:
+                return traceparent
+        except Exception:
+            logger.debug("traceparent does not follow version 00 specification")
```

Comment:
I will have to check this, because our internal documentation was mentioning to do it only for the 00 version. I am not sure what backwards compatibility will mean in terms of my regular expression, I will check the specification to make sure they are not gonna add more parts on it in any way.

Reply:
Here is what the W3C Trace Spec says:
> If a higher version is detected, the implementation SHOULD try to parse it by trying the following:
>
> * If the size of the header is shorter than 55 characters, the vendor should not parse the header and should restart the trace.
> * Parse trace-id (from the first dash through the next 32 characters). Vendors MUST check that the 32 characters are hex, and that they are followed by a dash (-).
> * Parse parent-id (from the second dash at the 35th position through the next 16 characters). Vendors MUST check that the 16 characters are hex and followed by a dash.
> * Parse the sampled bit of flags (2 characters from the third dash). Vendors MUST check that the 2 characters are either at the end of the string or followed by a dash.
> If all three values were parsed successfully, the vendor should use them.

https://www.w3.org/TR/trace-context/#versioning-of-traceparent